trackintel/analysis/tracking_quality.py

```diff
@@ -87,23 +92,27 @@ def temporal_tracking_quality(source, granularity="all", max_iter=60):
         # get the tracked day relative to the first day
         start_date = df["started_at"].min().date()
         df["day"] = df["started_at"].apply(lambda x: (x.date() - start_date).days)
+        dict_day_to_date = dict(zip(df["day"], df["started_at"].apply(lambda x: x.date())))
 
         # calculate per-user per-day raw tracking quality
         raw_quality = df.groupby(["user_id", "day"], as_index=False).apply(_get_tracking_quality_user, granularity)
         # add quality = 0 records
         quality = _get_all_quality(df, raw_quality, granularity)
+        quality["date"] = quality["day"].map(dict_day_to_date)
 
     elif granularity == "week":
         # split records that span several days
         df = _split_overlaps(df, granularity="day", max_iter=max_iter)
         # get the tracked week relative to the first day
         start_date = df["started_at"].min().date()
         df["week"] = df["started_at"].apply(lambda x: (x.date() - start_date).days // 7)
+        dict_week_num = dict(zip(df["week"], df["started_at"].apply(lambda x: pd.to_datetime(x).week)))
```

Comment:
I have a warning raised here:

> FutureWarning: weekofyear and week have been deprecated, please use DatetimeIndex.isocalendar().week instead, which returns a Series.  To exactly reproduce the behavior of week and weekofyear and return an Index, you may call pd.Int64Index(idx.isocalendar().week)

Also, the week_number is in float format, I think it should be int

Reply:
Warning removed, week number changed to `int`.